vision/convmixer_cifar10/convmixer.jl

```diff
@@ -0,0 +1,131 @@
+using Flux, MLDatasets
+using Flux: onehotbatch, onecold, DataLoader, Optimiser
+using BSON:@save,@load
+
+
+function ConvMixer(in_channels, kernel_size, patch_size, dim, depth, N_classes)
+    f = Chain(
+            Conv((patch_size, patch_size), in_channels=>dim, gelu; stride=patch_size),
+            BatchNorm(dim),
+            [
+                Chain(
+                    SkipConnection(Chain(Conv((kernel_size,kernel_size), dim=>dim, gelu; pad=SamePad(), groups=dim), BatchNorm(dim)), +),
+                    Chain(Conv((1,1), dim=>dim, gelu), BatchNorm(dim))
+                ) 
+                for i in 1:depth
+            ]...,
+            AdaptiveMeanPool((1,1)),
+            flatten,
+            Dense(dim,N_classes)
+        )
+    return f
+end
+
+function get_data(batchsize; dataset = MLDatasets.CIFAR10, idxs = nothing)
+    """
+    idxs=nothing gives the full dataset, otherwise (for testing purposes) only the 1:idxs elements of the train set are given.
+    """
+    ENV["DATADEPS_ALWAYS_ACCEPT"] = "true" 
+
+    # Loading Dataset
+    if idxs==nothing
+        xtrain, ytrain = dataset.traindata(Float32)
+        xtest, ytest = dataset.testdata(Float32)
+	else
+        xtrain, ytrain = dataset.traindata(Float32, 1:idxs)
+        xtest, ytest = dataset.testdata(Float32, 1:Int(idxs/10))
+    end
+
+    # Reshape Data to comply to Julia's (width, height, channels, batch_size) convention in case there are only 1 channel (eg MNIST)
+    if ndims(xtrain)==3
+        w = size(xtrain)[1]
+        xtrain = reshape(xtrain, (w,w,1,:))
+        xtest = reshape(xtest, (w,w,1,:))
+    end
+    
+    ytrain, ytest = onehotbatch(ytrain, 0:9), onehotbatch(ytest, 0:9)
+
+    train_loader = DataLoader((xtrain, ytrain), batchsize=batchsize, shuffle=true)
+    test_loader = DataLoader((xtest, ytest), batchsize=batchsize)
+
+    return train_loader, test_loader
+end
+
+function ℓ(dataloader, model, device)
+    
+    n = 0
+    loss = 0.0f0
+    acc = 0.0f0
+
+    for (x,y) in dataloader
+        x,y = x |> device, y |> device
```

Comment:
This doesn't seem efficient, could we move the data prior to the loading? Ie create a DataLoader with CuArrays instead of Array?

Reply:
I'm not sure if all GPUs people might want to try this out on would be able to support the model and all of CIFAR-10. One approach you could try is wrapping the dataloader in [`CUDA.CuIterator`](https://cuda.juliagpu.org/stable/usage/memory/#Batching-iterator), which handles this kind of memory management more efficienty.

Reply:
That would be roughly the same as the current version. Cifar isn't a very heavy dataset, so worse case might be tweaking the batch size.

Reply:
I had a look, and of the [two](https://github.com/FluxML/model-zoo/blob/master/vision/vgg_cifar10/vgg_cifar10.jl) [other](https://github.com/FluxML/model-zoo/blob/master/tutorials/60-minute-blitz/60-minute-blitz.jl) models using CIFAR-10, one uploads everything to GPU and the other works in batches. So as long as VRAM usage stays under, say, 4-6GB with the default hyperparams here, I'm fine with this.

Reply:
Maybe, for CIFAR10, we could have a dataloader with `CuArrays`, but this is clearly not doable for any other heavier model. AFAIK, it's really unusual (and often, impossible) to load a whole training dataset to GPU, no?